daisy_workflows/image_import/import_image.sh

```diff
@@ -48,24 +54,50 @@ function resizeDisk() {
   fi
 }
 
-# Mount GCS bucket containing the disk image.
-mkdir -p /gcs/${SOURCEBUCKET}
-gcsfuse --implicit-dirs ${SOURCEBUCKET} /gcs/${SOURCEBUCKET}
+function copyImageToScratchDisk() {
+  local sourceSizeGigabytes="$((1 + (SOURCE_SIZE_BYTES - 1) / BYTES_1GB))"
+  local scratchDiskSizeGigabytes=sourceSizeGigabytes
+  # We allocate double capacity for OVA, which would
+  # require making an additional copy of its enclosed VMDK.
+  if [[ "${SOURCE_FILE_TYPE}" == "ova" ]]; then
+     scratchDiskSizeGigabytes=$((sourceSizeGigabytes * 2))
+  fi
+
+  # This disk is initially created with 10GB of space.
+  # Enlarge it if that's insufficient to hold the input image.
+  if [[ ${scratchDiskSizeGigabytes} -gt 10 ]]; then
+    resizeDisk "${SCRATCH_DISK_NAME}" "${scratchDiskSizeGigabytes}" "${ZONE}"
+  fi
+
+  local scratchPath="/daisy-scratch/$(dirname ${SOURCEPATH})/"
```

Comment:
i don't understand. why do we care about the dir part of the path to source disk file here. why not put the disk file we're touching to a predefined path eg /daisy-scratch/sourcedisk.ext , where we'd know to find it in the tar and convert steps below?

Reply:
There's subsequent logic that assumes the image will be in this path. That was the motivation I'd say.

Reply:
we control both ends, though. if there's not a need to use the uploaded filename locally, my suggestion is to simplify

Reply:
It prevents multiple scripts using duplicate path. The machine can be reused, if we want.